### PR TITLE
JDBCrequests

### PR DESCRIPTION
**Pull request by Yanetta**

### CODE REVIEW

File: src/main/java/workwithJDBC/OrderDaoImpl.java [hunk 1]
```diff
@@ -5,15 +5,84 @@
...
+        Set<Order> orders = new HashSet<>();
+        while (rs.next()) {
+            Order order = new Order();
+            order.setAmount(rs.getBigDecimal("amount"));
+            order.setQty(rs.getBigDecimal("qty"));
+            order.setOrderNum(rs.getBigDecimal("order_num"));
+            order.setOrderDate(rs.getDate("order_date"));
+            order.setMfr(rs.getString("mfr"));
+
+            PreparedStatement stmt1 = connection.prepareStatement("Select * from  products where product_id=?");
```
Comment (un-install) [inline src/main/java/workwithJDBC/OrderDaoImpl.java:27]: Sholud works. But you can pull out this string from cycle.

Comment (un-install) [reply src/main/java/workwithJDBC/OrderDaoImpl.java:27]: and some others

File: src/main/java/workwithJDBC/OrderDaoImpl.java [hunk 2]
```diff
@@ -5,15 +5,84 @@
...
+            order.setOrderDate(rs.getDate("order_date"));
+            order.setMfr(rs.getString("mfr"));
+
+            Product product = new Product();
+            product.setProductID(rs.getString("product_Id"));
+            product.setMfrID(rs.getString("mfr_Id"));
+            product.setDescription(rs.getString("description"));
+            product.setPrice(rs.getBigDecimal("price"));
+            product.setQtyOnHand(rs.getBigDecimal("qty_On_Hand"));
+            order.setProduct(product);
```
Comment (un-install) [inline src/main/java/workwithJDBC/OrderDaoImpl.java:70]: Seters is okay, but you can use constructors to not spred the entityes

File: src/main/java/workwithJDBC/OrderDaoImpl.java [hunk 3]
```diff
@@ -27,7 +96,7 @@ public  Order findOrderById (BigDecimal id) throws SQLException {
             order.setOrderNum(rs.getBigDecimal("order_num"));
             order.setProduct(null);
             order.setOrderDate(rs.getDate("order_date"));
```
Comment (un-install) [inline src/main/java/workwithJDBC/OrderDaoImpl.java:98]: I think you need to set full product, like in the methods above

File: src/test/java/TestOrderDaoImpl/OrderDaoImplTest.java [hunk 4]
```diff
@@ -6,25 +6,37 @@
...
-    public void testFindOrderByIdNull() throws SQLException {
-        Order order = orderDao.findOrderById(BigDecimal.valueOf(3));
-        System.out.println(order);
-        assertNotNull(order);
-    }
+    public void testGetAllOrders() throws SQLException {
+        System.out.println("testGetAllOrders ____________________");
+        Set<Order> orders = orderDao.getAllOrders ();
+        System.out.println(orders);
+        assertTrue(orders.size() > 25);
```
Comment (un-install) [inline src/test/java/TestOrderDaoImpl/OrderDaoImplTest.java:31]: Bad testing parameter

Comment (un-install) [reply src/test/java/TestOrderDaoImpl/OrderDaoImplTest.java:31]: The table can and should has various size

File: src/test/java/TestOrderDaoImpl/OrderDaoImplTest.java [hunk 5]
```diff
@@ -6,25 +6,37 @@
...
 
+    }
+    @Test
+    public void testGetAllOrdersInnerJoin() throws SQLException {
+        System.out.println("testGetAllOrdersInnerJoin ____________________");
+        Set<Order> orders = orderDao.getAllOrdersInnerJoin ();
+        System.out.println(orders);
+        assertTrue(orders.size() > 25);
+    }
```
Comment (un-install) [inline src/test/java/TestOrderDaoImpl/OrderDaoImplTest.java:40]: Also

**Review by un-install — APPROVED** [review]

File: src/main/java/workwithJDBC/OrderDao.java [hunk 6]
```diff
@@ -6,8 +6,14 @@
...
-    Order findOrderById (BigDecimal id) throws SQLException;
-    boolean insertOrder (Order order)throws SQLException;
-    boolean updateOrder (Order order)throws SQLException;
-    boolean deleteOrder (BigDecimal id)throws SQLException;
+
+    Set<Order> getAllOrdersInnerJoin() throws SQLException;
+
+    Order findOrderById(BigDecimal id) throws SQLException;
+
+    boolean insertOrder(Order order) throws SQLException;
```
Comment (SerhiiPasko) [inline src/main/java/workwithJDBC/OrderDao.java:14]: Why did you write there are methods?

File: src/test/java/TestOrderDaoImpl/OrderDaoImplTest.java (hunk 4 above)
Comment (SerhiiPasko) [reply src/test/java/TestOrderDaoImpl/OrderDaoImplTest.java:31]: It is ok

File: src/test/java/TestOrderDaoImpl/OrderDaoImplTest.java (hunk 5 above)
Comment (SerhiiPasko) [reply src/test/java/TestOrderDaoImpl/OrderDaoImplTest.java:40]: Good test in our case - cheak that results in both methods are same

**Review by SerhiiPasko — APPROVED** [review]
Good. Read comments

**Pull request merged**